src/mono/mono/metadata/icall.c

```diff
@@ -2089,31 +2136,9 @@ ves_icall_RuntimeFieldInfo_SetValueInternal (MonoReflectionFieldHandle field, Mo
 			MonoGenericClass *gclass = type->data.generic_class;
 			g_assert (!gclass->context.class_inst->is_open);
 
-			if (mono_class_is_nullable (mono_class_from_mono_type_internal (type))) {
-				MonoClass *nklass = mono_class_from_mono_type_internal (type);
```

Comment:
Why is this removed ?

Reply:
The nullable boxing is no longer done here but from managed code via the Rebox{To/From}Nullable, from `RuntimeType.CheckValue`